packages/internal-test-utils/internalAct.js

```diff
@@ -63,6 +73,28 @@ export async function act<T>(scope: () => Thenable<T>): Thenable<T> {
   // public version of `act`, though we maybe should in the future.
   await waitForMicrotasks();
 
+  const errorHandlerDOM = function (event: ErrorEvent) {
+    // Prevent logs from reprinting this error.
+    event.preventDefault();
+    thrownErrors.push(event.error);
+  };
+  const errorHandlerNode = function (err: mixed) {
+    thrownErrors.push(err);
+  };
+  // We track errors that were logged globally as if they occurred in this scope and then rethrow them.
+  if (actingUpdatesScopeDepth === 1) {
+    if (
+      typeof window === 'object' &&
+      typeof window.addEventListener === 'function'
+    ) {
+      // We're in a JS DOM environment.
+      window.addEventListener('error', errorHandlerDOM);
+    } else if (typeof process === 'object') {
+      // Node environment
+      process.on('uncaughtException', errorHandlerNode);
```

Comment:
Resuming operation after `uncaughtException` is not safe with Node.js

https://nodejs.org/api/process.html#warning-using-uncaughtexception-correctly

> The 'uncaughtException' event is emitted when an uncaught JavaScript exception bubbles all the way back to the event loop. By default, Node.js handles such exceptions by printing the stack trace to stderr and exiting with code 1, overriding any previously set [process.exitCode](https://nodejs.org/api/process.html#processexitcode_1). Adding a handler for the 'uncaughtException' event overrides this default behavior

This means that we users no longer exit on `uncaughtException`, which is overriding default behaviour. `errorHandlerNode` should call `process.exit(1)`.

Reply:
This is not a public API